src/main/java/net/mcreator/blockly/java/blocks/ProcedureCallBlock.java

```diff
@@ -42,48 +40,50 @@ public class ProcedureCallBlock implements IBlockGenerator {
 		String type = block.getAttribute("type");
 
 		if (element != null && !"".equals(element.getTextContent())) {
+			int depCount = 0;
+			Map<Integer, String> names = new HashMap<>();
+			Map<Integer, String> args = new HashMap<>();
+			Element mutation = XMLUtil.getFirstChildrenWithName(block, "mutation");
+			if (mutation != null && mutation.hasAttribute("inputs")
+					&& !mutation.getAttribute("inputs").equals("undefined")) {
+				depCount = Integer.parseInt(mutation.getAttribute("inputs"));
+				Map<String, Element> fields = XMLUtil.getChildrenWithName(block, "field").stream()
+						.filter(e -> e.getAttribute("name").matches("name\\d+"))
+						.collect(Collectors.toMap(e -> e.getAttribute("name"), e -> e));
+				Map<String, Element> inputs = XMLUtil.getChildrenWithName(block, "value").stream()
+						.filter(e -> e.getAttribute("name").matches("arg\\d+"))
+						.collect(Collectors.toMap(e -> e.getAttribute("name"), e -> e));
+				for (int i = 0; i < depCount; i++) {
+					names.put(i, fields.remove("name" + i).getTextContent());
+					if (inputs.containsKey("arg" + i)) {
+						args.put(i, BlocklyToCode.directProcessOutputBlock(master, inputs.remove("arg" + i)));

```

Comment:
Here you could call `master.directProcessOutputBlockWithoutParentheses` instead

Reply:
✔️ Fixed